modules/openapi-generator/src/main/resources/python/model_utils.mustache

```diff
@@ -910,6 +910,13 @@ def deserialize_model(model_data, model_class, path_to_item, check_type,
                    _configuration=configuration,
                    _spec_property_naming=spec_property_naming)
 
+    # An attribute named `self` received from the api will conflicts with the reserved `self`
+    # parameter of a class method. During generation, `self` attributes are mapped
+    # to `_self` in models. Here, we map it directly.
+    if "self" in input_value:
+        input_value["_self"] = input_value["self"]
+        del input_value["self"]
```

Comment:
So I am concerned that this code is manually handling one case only.
Why does the convert_js_args_to_python_args fn not correctly handle this parameter?
That fn is used to decorate the init method: https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python/petstore_api/model/pet.py#L120

Fn definition is here:
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/python/petstore_api/model_utils.py#L1559

Which calls change_keys_js_to_python
https://github.com/OpenAPITools/openapi-generator/blob/46326249d2c9df8ec530199ea6dbb8821acf1dd1/samples/openapi3/client/petstore/python/petstore_api/model_utils.py#L1041
Which uses model_class.attribute_map to convert to the spec property name.
If the property name is self and the property is optional, it is passed in kwargs.

Reply:
Hmm this is happening because the kwargs self parameter is assigned to the self argument of the model:
```
>>> class A:
...   def __init__(self, **kwargs):
...     self.kwargs = kwargs
... 
>>> 
>>> A()
<__main__.A object at 0x10673fdc0>
>>> kwargs = dict(self=5)
>>> kwargs
{'self': 5}
>>> A(**kwargs)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got multiple values for argument 'self'
```

Reply:
In fact this is my error:
```
  File ".../model_utils.py", line 1287, in attempt_convert_item
    return deserialize_model(input_value, valid_class,
  File ".../model_utils.py", line 1201, in deserialize_model
    return model_class(**kw_args)
TypeError: wrapped_init() got multiple values for argument 'self'
```
So `self` is assigned as a decorator parameter and not as the kwargs to process.

Reply:
Other solution would be to deviate from python standard and set the self parameter to something highly improbable to set as an attribute in an api

```
def convert_js_args_to_python_args(fn):
    from functools import wraps

    @wraps(fn)
    def wrapped_init(__wrapped_self__no_collision, *args, **kwargs):
        spec_property_naming = kwargs.get("_spec_property_naming", False)
        if spec_property_naming:
            kwargs = change_keys_js_to_python(
                kwargs, __wrapped_self__no_collision.__class__
            )
        return fn(__wrapped_self__no_collision, *args, **kwargs)

    return wrapped_init
```

Reply:
> set the self parameter to something highly improbable
Yes, that's why my asked solution is to rename it to `_self`. One would need to change wrapped_init self to `_self` and the model `__init__`  self to `_self` This prevents the collision